src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,332 @@
+
+  private TalonFX pivotMotor;
+  private TalonFX conveyorMotor;
+  private Angle pivotSetpoint = kPivotRetractAngle;
+
+  private DutyCycleEncoder throughBoreEncoder;
+
+  // Sensor closest to birdhouse
+  private TOFSensorSimple sensor;
+  private SimDeviceSim simSensor;
+  private SimBoolean simbeam;
+
+  private Mechanism2d mech = new Mechanism2d(2, 2);
+  private ComplexWidget widget;
+  private MechanismLigament2d pivotArmMech;
+
+  private enum IntakeState {
+    Idle,
+    atSensor,
+    coralLoaded
+  }
+
+  private IntakeState state = IntakeState.Idle;
+  private Timer timer = new Timer();
+
+  private SingleJointedArmSim pivotSim;
+
+  private DebugEntry<Double> pivotOutput;
+  private DebugEntry<String> currentCommand;
+
+  public IntakeSubsystem() {
+    intakeMotor = new TalonFX(CANIDs.kIntakeMotor);
+    pivotMotor = new TalonFX(CANIDs.kPivotMotor);
+    conveyorMotor = new TalonFX(CANIDs.kConveyorMotor);
+    sensor = new TOFSensorSimple(CANIDs.kConveyorSensor, Inches.of(1), TOFType.LASER_CAN);
+    throughBoreEncoder = new DutyCycleEncoder(DIOConstants.kthroughBoreEncoderID, 1, armZero);
+
+    /**
+     * Once the gains are configured, the Position closed loop control request can be sent to the
+     * TalonFX. The control request object has an optional feedforward term that can be used to add
+     * an arbitrary value to the output, which can be useful to account for the effects of gravity
+     * or friction.
+     *
+     * <p>// create a position closed-loop request, voltage output, slot 0 configs final
+     * PositionVoltage m_request = new PositionVoltage(0).withSlot(0);
+     *
+     * <p>// set position to 10 rotations m_talonFX.setControl(m_request.withPosition(10));
+     */
+    var slot0Configs = new Slot0Configs();
+    slot0Configs.kP = kPivotP;
+    slot0Configs.kI = kPivotI;
+    slot0Configs.kD = kPivotD;
+    slot0Configs.kG = kPivotG;
+    slot0Configs.kA = kPivotA;
+    slot0Configs.kV = kPivotV;
+    slot0Configs.GravityType = kPivotGravityType;
+
+    var feedbackConfigs = new FeedbackConfigs();
+    feedbackConfigs.RotorToSensorRatio = 1;
+    feedbackConfigs.SensorToMechanismRatio = kSensorToMechanism;
+
+    var pivotMotionMagic =
+        new MotionMagicConfigs()
+            .withMotionMagicCruiseVelocity(kMotionMagicCruiseVelocity)
+            .withMotionMagicAcceleration(kMotionMagicAcceleration)
+            .withMotionMagicJerk(kMotionMagicJerk);
+
+    pivotMotor.getConfigurator().apply(slot0Configs);
+    pivotMotor.getConfigurator().apply(feedbackConfigs);
+    pivotMotor.getConfigurator().apply(pivotMotionMagic);
+    pivotMotor.setControl(new CoastOut());   // Temporary
+
+    pivotOutput = new DebugEntry<Double>(0.0, "Pivot Output", this);
+    currentCommand = new DebugEntry<String>("none", "Pivot Command", this);
+
+    if (throughBoreEncoder.get() != kPivotRetractAngle.times(kGearing).in(Rotations)) {
+      // pivotMotor.setPosition(kPivotRetractAngle);
+    }
```

Comment:
You should seed the position to the motor from the through bore encoder here. 
You will need to factor in the gear ratio and need to add an offset based on how the through bore encoder is oriented.